src/figuresClasses.ts

```diff
@@ -1,11 +1,97 @@
-export interface Figure {}
+enum Shapes {
+  circle = 'circle',
+  triangle = 'triangle',
+  rectangle = 'rectangle',
+}
+
+enum Colors {
+  red = 'red',
+  blue = 'blue',
+  green = 'green',
+}
+
+export interface Figure {
+  shape: Shapes;
+  color: Colors;
+  getArea(): number;
+}
+
+export class Triangle implements Figure {
+  shape = Shapes.triangle;
+
+  constructor(
+    public color: Colors,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (this.a <= 0 || this.b <= 0 || this.c <= 0) {
+      throw new Error(
+        'Invalid data, the value of the side of the triangle cannot be negative',
+      );
+    }
+
+    if (
+      this.a + this.b <= this.c ||
+      this.a + this.c <= this.b ||
+      this.b + this.c <= this.a
+    ) {
+      throw new Error('Incorrect value(s) for triangle');
```

Comment:
[CODE STYLE] - If you throw an error try to provide its detailed description in the error message. The current message 'Incorrect value(s) for triangle' could be more descriptive by explaining the mathematical condition that must be met for the triangle sides.